logstash-core-event/spec/logstash/event_spec.rb

```diff
@@ -44,6 +44,11 @@
         subject["@metadata"] = { "action" => "index" }
         expect(subject["[@metadata][action]"]).to eq("index")
       end
+
+      it "should add key when setting nil value" do
```

Comment:
yeah, I will do specs reunification pass at some point. actually, I'll create an issue for it (did I create one for this already? memory #fail) - there is a bit of duplication and what is really generic should be in core-event and java specific only in core-event-java.